src/handle/tactic.ml

```diff
@@ -195,26 +205,28 @@ let handle : Sig_state.t -> bool -> proof_state -> p_tactic -> proof_state =
   | P_tac_solve -> assert false (* done before *)
   | P_tac_apply pt ->
       let p = new_problem() in
```

Comment:
What is this p for if scoping does not create metavariables anymore? Problem creation should be done as close as possible to their use.

Reply:
It seems that problem creation could be moved to tac_refine.

Reply:
Other tactics give problem with metavariables to `tac_refine`, it doesn't look obvious.

Reply:
Tactics `induction` and `have` provide a non empty problem to `tac_refine`.